### ISSUE TITLE
setTimeout used instead of requestAnimationFrame

### ISSUE THREAD

**Issue by davidstellini** [glidejs/glide#655]
I was looking at a burndown chart and saw the carousel was responsible for a large chunk of JS execution time:
<img width="557" alt="image" src="https://user-images.githubusercontent.com/4324419/226920829-8d247cb5-a5e4-4f1a-968d-cd79cf772215.png">

When looking at the culprit, it seems like most of the time is spent here:
https://github.com/glidejs/glide/blob/5e958de5e9a436916b3e7411384ff92d73b0cf17/src/components/transition.js#L56

I searched the repo for requestAnimationFrame and didn't find anything, could you kindly confirm if this is a problem or not? I can open a pull request for it.

**Comment by jedrzejchalubek** [comment]
Yeah, basing this on `requestAnimationFrame` would be much better. setTimeout lacks performance and precision here
